bluesky/utils.py

```diff
@@ -1486,3 +1486,54 @@ def _rearrange_into_parallel_dicts(readings):
         data[key] = payload['value']
         timestamps[key] = payload['timestamp']
     return data, timestamps
+
+
+def plot_prune_fifo(bec, n, y, x):
+    """
+    find the plot with axes x and y and replot with only the last *n* lines
+
+    Note: this is not a bluesky plan.  Call it as normal Python function.
+
+    EXAMPLE::
+
+        plot_prune_fifo(bec, 1, noisy, m1)
+
+    PARAMETERS
+    
+    bec : object
+        instance of BestEffortCallback
+    
+    n : int
+        number of plots to keep
+    
+    y : object
+        instance of ophyd.Signal (or subclass), 
+        dependent (y) axis
+    
+    x : object
+        instance of ophyd.Signal (or subclass), 
+        independent (x) axis
+    """
+    assert n >= 0, "n must be 0 or greater"
+    for liveplot in bec._live_plots.values():
+        lp = liveplot.get(y.name)
+        if lp is None or lp.x != x.name or lp.y != y.name:
+            continue
+
+        # pick out only the traces that contain plot data
+        # skipping the lines that show peak centers
+        lines = [
+            tr
+            for tr in lp.ax.lines
+            if len(tr._x) != 2 
```

Comment:
Why not the `>` condition? Do we care about the len=0 or 1?

Reply:
We're trimming out the lines which mark the peak centers, specifically.  Other cases can take care of themselves, as usual.  The logic is correct for this intent.

Reply:
The lines which mark the peak centers have exactly two points and the X value is identical for each of those.  It is truly rare (although possible) to have measured data with exactly two points that have the same X.  Not sure I have any other means to identify.  Perhaps if I look at the Y values?  They might have special meaning, as well, and that special meaning could be used to identify them uniquely.

Reply:
Looks OK for now.